main.go

```diff
@@ -182,53 +190,69 @@ func main() {
 		log.Errorf("Test task failed, error: %v", testErr)
 	}
 	fmt.Println()
-
-	log.Infof("Export reports:")
+	log.Infof("Export HTML results:")
 	fmt.Println()
 
-	reports, err := getArtifacts(gradleProject, started, config.ReportPathPattern, true, true)
+	reports, err := getArtifacts(gradleProject, started, config.HTMLResultDirPattern, true, true)
 	if err != nil {
 		failf("Failed to find reports, error: %v", err)
 	}
 
-	if err := exportArtifacts(deployDir, reports); err != nil {
+	if err := exportArtifacts(config.DeployDir, reports); err != nil {
 		failf("Failed to export reports, error: %v", err)
 	}
 
 	fmt.Println()
-
-	log.Infof("Export results:")
+	log.Infof("Export XML results:")
 	fmt.Println()
 
-	results, err := getArtifacts(gradleProject, started, config.ResultPathPattern, true, true)
+	results, err := getArtifacts(gradleProject, started, config.XMLResultDirPattern, true, true)
 	if err != nil {
 		failf("Failed to find results, error: %v", err)
 	}
 
-	if err := exportArtifacts(deployDir, results); err != nil {
+	if err := exportArtifacts(config.DeployDir, results); err != nil {
 		failf("Failed to export results, error: %v", err)
 	}
 
-	log.Infof("Export test results for test addon:")
-	fmt.Println()
+	if config.TestResultDir != "" {
+		// Test Addon is turned on
+		fmt.Println()
+		log.Infof("Export XML results for test addon:")
+		fmt.Println()
 
-	resultXMLs, err := getArtifacts(gradleProject, started, resultArtifactPathPattern, false, false)
-	if err != nil {
-		log.Warnf("Failed to find test result XMLs, error: %s", err)
-	} else {
-		if baseDir := os.Getenv("BITRISE_TEST_RESULT_DIR"); baseDir != "" {
+		lastOtherDirIdx := -1
+		xmlResultFilePattern := config.XMLResultDirPattern
+		if !strings.HasSuffix(xmlResultFilePattern, "*.xml") {
```

Comment:
Optionally would add that the whole directory is exported as artifacts and xml files are looked up for the test report addon  in the input description.